### PR TITLE
Add prototype poisoning doc

### PR DESCRIPTION
**Pull request by jsumners**
Resolves issue #3003.

cc: @hueniverse

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: docs/Guides/Prototype-Poisoning.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+While the issue at hand was about incoming request payloads, we had to pause and
+check if it could also impact data coming via the query string, cookies, and
+headers. Basically, anything that gets serialized into objects from text.
+
+We quickly confirmed node default query string parser was fine as well as its
+header parser. I identified one potential issue with base64-encoded JSON cookies
+as well as the usage of custom query string parsers. We also wrote some tests to
+confirm that the most popular third-party query string parser  —
+[qs ](https://www.npmjs.com/package/qs) —  was not vulnerable (it is not!).
```
Comment (luisorbaiceta) [inline docs/Guides/Prototype-Poisoning.md:182]: There is an extra space that will get underlined on website links

**Review by luisorbaiceta — CHANGES_REQUESTED** [review]

File: docs/Guides/Prototype-Poisoning.md (hunk 1 above)
Comment (jsumners) [inline docs/Guides/Prototype-Poisoning.md:182]:
```suggestion
[qs](https://www.npmjs.com/package/qs) —  was not vulnerable (it is not!).
```

**Review by luisorbaiceta — APPROVED** [review]
lgtm!

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by hueniverse** [comment]
I think you should remove all the non-technical relevant stuff and instead of the strong attribution, simply replace it with "Based on the article ... by Eran Hammer". I'm giving you full permission to use this text as you find useful. Don't be shy to change it.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by jsumners** [comment]
> I think you should remove all the non-technical relevant stuff and instead of the strong attribution, simply replace it with "Based on the article ... by Eran Hammer". I'm giving you full permission to use this text as you find useful. Don't be shy to change it.

Thank you. I appreciate the suggestion. Right now, I do not have the time to massage the language like that. I have created a new issue to address it in the future.

**Comment by rluvaton** [comment]
Just wanted to say it was delightful to read, thank you 😀

**Pull request merged**